src/templates/utils-common.js

```diff
@@ -214,8 +214,21 @@ export function setLocaleCookie (locale, res, { useCookie, cookieDomain, cookieK
     }
 
     const redirectCookie = cookieSerialize(cookieKey, locale, cookieOptions)
-    headers.push(redirectCookie)
+    let foundRedirectCookie = false
+    const cookies = headers.map(header => {
+      const cookie = cookieParse(header)
+      if (cookie[cookieKey]) {
+        cookie[cookieKey] = locale
```

Comment:
There is no need to set the cookie value if it's being dropped anyway.

Though I would prefer to have a filter that just filters out existing cookie and then always adds the new cookie after filtering is done.

Reply:
I just forgot to remove `cookie[cookieKey] = locale`

The points are addressed.